src/app/core/components/header/header.component.ts

```diff
@@ -1,8 +1,22 @@
 import { Component } from '@angular/core';
 
+import { Observable } from 'rxjs';
+
+import { AuthService } from '@app/services/auth/auth.service';
+
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: [ './header.component.scss' ],
 })
-export class HeaderComponent { }
+export class HeaderComponent {
+  constructor(private authService: AuthService) { }
+
+  get showUserPanel$(): Observable<boolean> {
+    return this.authService.isAuthenticated$;
+  }
+
+  get userName$(): Observable<string> {
+    return this.authService.userName$;
+  }
```

Comment:
please, move into the `app-user-panel`

Reply:
so, just clean it up. All displaying logic and service injection must be in the component. No need to duplicate it here.

Reply:
Ok